### ISSUE TITLE
Module Request: Ross XPression

### ISSUE THREAD

**Issue by spiritimk** [bitfocus/companion#440]
**Describe the feature**
Add module to trigger Ross XPression motion graphics from Companion.
https://www.rossvideo.com/products-services/acquisition-production/cg-graphics-systems/xpression/

**Is this platform dependent (windows, mac, ..)?**
Platform Independent.
Requires support for TCP sockets.

**If documentation is required to implement, do you know where to find it?**
Protocol consists in tcp socket connections to port 7788 of XPression machine.
Documentation of the protocol can be found at http://help.rossvideo.com/carbonite-device/Topics/Protocol/RossTalk/XPN/RT-XPN-Comm.html#topic.RT-XPN-Comm

**Usecases**
XPression is one of the industry standards for realtime motion graphics. For example it is used in ESL (Electronic Sports League, https://www.eslgaming.com/) for all online and on site broadcasts to handle dynamic graphics, from small studio to big productions like Intel Extreme Masters.

**Comment by JeffreyDavidsz** [comment]
@spiritimk there's a RossTalk module which has the commands for Expression. I've added some text in the Help.md but it needs to be merged first. For now just try that one!

**Comment by paddy2me** [comment]
@willosof I hate to be picky, but the product is **XPression**, not _Expression_. _Expression_ is a software product from Microsoft™, while **XPression** is a real-time motion graphics engine from Ross Video. It would be great if you could revise the product name in your listings in the application, so that users can find it.
Thanks,
Patrick Twomey
Director of XPression Product Marketing

**Comment by JeffreyDavidsz** [comment]
Pretty sure I did the help file, I'll adjust it

**Comment by paddy2me** [comment]
WHen you go to create a new instance, it is titled Expression
![image](https://user-images.githubusercontent.com/47610458/52740885-0d8c0b80-2fa2-11e9-8893-4977bb241450.png)

**Comment by JeffreyDavidsz** [comment]
@paddy2me it is already adjusted in the latest build

**Comment by paddy2me** [comment]
Thank you!

**Comment by willosof** [comment]
I love the work you put into the screenshot, btw! ;-)

**Comment by paddy2me** [comment]
Sorry. As a marketing guy, that kind of stuff takes me a second to do. The goal is to make everything clear so it only needs to be addressed once.
Now, if I could just get this Stream Desk to talk to XPression, I would be a happy guy. Their app and your tools allow me to program the Stream Deck, but the Stream Deck doesn't seem to be able to reach out to XPression.

**Comment by JeffreyDavidsz** [comment]
Denpending on firmware, quit the Streamdeck app. Start companion and add the module, in config set the IP address of the XPression. You should be able to use the commands. Make sure you can ping from your machine

**Comment by paddy2me** [comment]
ALso, found this one...
![image](https://user-images.githubusercontent.com/47610458/52901748-57a90300-31d5-11e9-922f-afa285f06135.png)

**Comment by paddy2me** [comment]
...and here
![image](https://user-images.githubusercontent.com/47610458/52901887-5082f480-31d7-11e9-8097-cb35c4b28af0.png)

**Comment by paddy2me** [comment]
> Depending on firmware, quit the Streamdeck app. Start companion and add the module, in config set the IP address of the XPression. You should be able to use the commands. Make sure you can ping from your machine

I still can not get this to work on a _WIN10_ machine. I can ping this, and send the same commands from Ross Video Dashboard or HyperTerm. Elgato Stream Deck app and Companion are not able to set the **RossTalk** to **XPression** on the same computer. Tried _127.0.0.1_ as well as _192.168.1.241_, which is the local IP on my LAN. Not at all faulting you guys, this is just something that isn't working out of the box, and I don't have time to play with it anymore.

Gonna take these items back to Best Buy for now.

**Comment by JeffreyDavidsz** [comment]
You'll need to download the latest build, not the official 1.3.0 in order to see the changes

**Comment by JeffreyDavidsz** [comment]
To bad you gave up. We would have gladly helped you

**Comment by paddy2me** [comment]
Well... I still have them. I am willing to continue on with the testing. ;-)


I am not a programmer, and I can't distract the XPression Development team for this. It might seem like it would be a small task for them, but they have thousands of small tasks that are way ahead of this.

**Comment by JeffreyDavidsz** [comment]
Ok, start with quiting the Streamdeck app from elgato, you don't need it for now. Go to builds.bitfocus.io and download the latest (in companion directory) Then start companion. Interface can be on default 127.0.0.1 and port 8000. Launch GUI and create instance for rosstalk. Now what happened is that we wanted to create a separate module for XPression. But when reading the specs it showed that almost all standard commands where in the RossTalk module already. So this hasn't been tested on a actual machine (as far as I know). And it could be possible that XPression needs to accept incoming command? If this still doesn't work for you I can ask one of the switcher owners who already use companion to test it.

**Comment by paddy2me** [comment]
Ok. 
First off, thank you!
Second, I downloaded the latest version for Win64, and that is now working!
Third, I saw all the corrections to the word XPression. Thank you for that, as well. We chose something cliche for the name, and we try hard to enforce it. When it is done properly, we smile... alot.
Give me some time to play with this, and i will create something to share with our customer base and point it back to your software/site when approved by you.

Thanks again!
